pkg/network/setup/filters.go

```diff
@@ -0,0 +1,53 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors
+ *
+ */
+
+package network
+
+import (
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
+)
+
+func FilterNetsForVMStartup(vmi *v1.VirtualMachineInstance) []v1.Network {
+	nonAbsentIfaces := vmispec.FilterInterfacesSpec(vmi.Spec.Domain.Devices.Interfaces, func(iface v1.Interface) bool {
+		return iface.State != v1.InterfaceStateAbsent
+	})
+
+	return vmispec.FilterNetworksByInterfaces(vmi.Spec.Networks, nonAbsentIfaces)
+}
+
+func FilterNetsForLiveUpdate(vmi *v1.VirtualMachineInstance) []v1.Network {
```

Comment:
Sounds better thanks for changing it, I would still pick some shorter yet meaning full name, how about FilterUpdatedNetworks?

Reply:
I think it is ambiguous since one cannot understand from the context what has changed / updated.

Reply:
You cant understand what has changed by the current name anyways.

Please rename this function so it doesn't describe the context from where it should be called (i.e: "ForLiveUpdate").
The context should be realized from the caller; it called at virt-handler VM controller, on the update flow that should do.
In general having the name not mentioning the context makes this function more portable.

Reply:
Same as https://github.com/kubevirt/kubevirt/pull/13784#discussion_r1935435394.